File: app/src/main/java/com/tustar/demo/BaseActivity.java
package com.tustar.demo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by tustar on 4/10/16.
 */
public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }
}

File: core/ui/src/main/java/com/tustar/ui/BaseActivity.kt
package com.tustar.ui

import android.os.Bundle
import android.view.LayoutInflater
import androidx.viewbinding.ViewBinding
import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity<T : ViewBinding> : AppCompatActivity() {

    protected lateinit var binding: T
    protected abstract val bindingInflate: (LayoutInflater) -> T

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = bindingInflate(layoutInflater)
        setContentView(binding.root)
        initViews()
        initData()
    }

    abstract fun initViews()

    abstract fun initData()
}